internal/cmd/context/ssh_key.go

```diff
@@ -0,0 +1,211 @@
+
+import (
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+	"sync"
+
+	"github.com/spf13/cobra"
+
+	"github.com/hetznercloud/cli/internal/cmd/cmpl"
+	"github.com/hetznercloud/cli/internal/state"
+	"github.com/hetznercloud/cli/internal/state/config"
+	"github.com/hetznercloud/hcloud-go/v2/hcloud"
+)
+
+func newSSHKeyCommand(s state.State) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "ssh-key",
+		Short:                 "Manage a context's default SSH key",
+		Args:                  cobra.NoArgs,
+		TraverseChildren:      true,
+		DisableFlagsInUseLine: true,
+	}
+	cmd.AddCommand(newSSHKeyAddCommand(s))
+	cmd.AddCommand(newSSHKeyRemoveCommand(s))
+	cmd.AddCommand(newSSHKeyListCommand(s))
+	return cmd
+}
+
+func newSSHKeyAddCommand(s state.State) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "add SSH-KEY...",
+		Short:                 "Add a default SSH key to the context",
+		TraverseChildren:      true,
+		DisableFlagsInUseLine: true,
+		RunE:                  state.Wrap(s, runSSHKeyAdd),
+	}
+	cmd.Flags().String("context", "", "Name of the context to add the default SSH key(s) to")
+	_ = cmd.RegisterFlagCompletionFunc("context", cmpl.SuggestCandidates(config.ContextNames(s.Config())...))
+
+	cmd.Flags().Bool("all", false, "Add all available SSH keys to the context")
+	_ = cmd.RegisterFlagCompletionFunc("all", cmpl.SuggestCandidates("true", "false"))
+	return cmd
+}
+
+func runSSHKeyAdd(s state.State, cmd *cobra.Command, args []string) error {
+
+	ctx, err := getContext(s, cmd)
+	if err != nil {
+		return err
+	}
+
+	s.Client().WithOpts(hcloud.WithToken(ctx.Token))
+	keys := args
+
+	all, _ := cmd.Flags().GetBool("all")
+	if all {
+		allKeys, err := s.Client().SSHKey().All(s)
+		if err != nil {
+			return err
+		}
+		if len(allKeys) == 0 {
+			return fmt.Errorf("no SSH keys available")
+		}
+		keys = make([]string, len(allKeys))
+		for i, key := range allKeys {
+			keys[i] = fmt.Sprintf("%d", key.ID)
+		}
+	} else {
+		if len(keys) == 0 {
+			return fmt.Errorf("no SSH keys specified")
+		}
+		var (
+			notExist []string
+			wg       = &sync.WaitGroup{}
+			mu       = &sync.Mutex{}
+		)
+		wg.Add(len(keys))
+		for _, key := range keys {
+			key := key
+			go func() {
+				k, _, _ := s.Client().SSHKey().Get(s, key)
+				if k == nil {
+					mu.Lock()
+					notExist = append(notExist, key)
+					mu.Unlock()
+				}
+				wg.Done()
+			}()
+		}
+		wg.Wait()
+		if len(notExist) > 0 {
+			_, _ = fmt.Fprintf(os.Stderr, "Warning: The given SSH key(s) %s do not exist in context \"%s\"\n", strings.Join(notExist, ", "), ctx.Name)
+		}
+	}
+
+	ctx.SSHKeys = append(ctx.SSHKeys, keys...)
+	// remove duplicates
+	slices.Sort(ctx.SSHKeys)
+	ctx.SSHKeys = slices.Compact(ctx.SSHKeys)
+
+	if err := s.Config().Write(); err != nil {
+		return err
+	}
+
+	cmd.Printf("Added SSH key(s) %s to context \"%s\"\n", strings.Join(keys, ", "), ctx.Name)
+	return nil
+}
+
+func newSSHKeyRemoveCommand(s state.State) *cobra.Command {
```

Comment:
Could you add autocomplete for `SSH-KEY` to this command?

Reply:
Done in d87cdc1